godel_plugins/src/widgets/robot_blending_widget.cpp

```diff
@@ -429,244 +491,126 @@ void RobotBlendingWidget::save_robot_scan_parameters()
-	tf::poseTFToMsg(tcp_to_cam_pose_widget_->get_values(),robot_scan_parameters_.tcp_to_cam_pose);
-
-}
-
-PoseWidget::PoseWidget(QWidget *parent):
-		QWidget(parent)
+void RobotBlendingWidget::simulate_motion_plan_handler()
 {
-	ui_.setupUi(this);
-	set_values(tf::Transform::getIdentity());
+	if (ui_.ListPathResults->currentItem() == NULL) return;
+	std::string name = ui_.ListPathResults->currentItem()->text().toStdString();
+	ROS_INFO_STREAM("Selected " << name << " to be simulated");
+	if (!name.empty()) select_motion_plan(name, true);
 }
 
-void PoseWidget::set_values(const geometry_msgs::Pose& p)
+void RobotBlendingWidget::execute_motion_plan_handler()
 {
-	tf::Transform t;
-	tf::poseMsgToTF(p,t);
-	set_values(t);
+	if (ui_.ListPathResults->currentItem() == NULL) return;
+	std::string name = ui_.ListPathResults->currentItem()->text().toStdString();
+	ROS_INFO_STREAM("Selected " << name << " to be executed");
+	if (!name.empty()) select_motion_plan(name, false);
 }
 
-void PoseWidget::set_values(const tf::Transform& t)
+void RobotBlendingWidget::save_motion_plan_handler()
 {
-	tf::Vector3 p = t.getOrigin();
-	tfScalar rx,ry,rz;
-	t.getBasis().getRPY(rx,ry,rz,1);
-	ui_.LineEditX->setText(QString::number(p.x()));
-	ui_.LineEditY->setText(QString::number(p.y()));
-	ui_.LineEditZ->setText(QString::number(p.z()));
-	ui_.LineEditRx->setText(QString::number(RAD2DEG(rx)));
-	ui_.LineEditRy->setText(QString::number(RAD2DEG(ry)));
-	ui_.LineEditRz->setText(QString::number(RAD2DEG(rz)));
+	QString filepath = QFileDialog::getSaveFileName(this, "Save Motion Plan");
+	// No file selected, return immediately
+	ROS_DEBUG_STREAM("You want to save motion plan to: " << filepath.toStdString());
+	request_load_save_motions(filepath.toStdString(), false);
 }
 
-tf::Transform PoseWidget::get_values()
+void RobotBlendingWidget::load_motion_plan_handler()
 {
-	double x,y,z,rx,ry,rz;
-	x = ui_.LineEditX->text().toDouble();
-	y = ui_.LineEditY->text().toDouble();
-	z = ui_.LineEditZ->text().toDouble();
-	rx = DEG2RAD(ui_.LineEditRx->text().toDouble());
-	ry = DEG2RAD(ui_.LineEditRy->text().toDouble());
-	rz = DEG2RAD(ui_.LineEditRz->text().toDouble());
-
-	// create transform
-	tf::Vector3 p(x,y,z);
-	tf::Quaternion q;
-	q.setRPY(rx,ry,rz);
-
-	return tf::Transform(q,p);
+	QString filepath = QFileDialog::getOpenFileName(this, "Load Motion Plan");
+	// No file selected, so return immediately
+	ROS_DEBUG_STREAM("You want to load a motion plan from: " << filepath.toStdString());
+	request_load_save_motions(filepath.toStdString(), true);
 }
 
-SurfaceDetectionConfigWidget::SurfaceDetectionConfigWidget(godel_msgs::SurfaceDetectionParameters params)
+void RobotBlendingWidget::request_load_save_motions(const std::string& path, bool isLoad)
 {
-	surface_detection_parameters_ = params;
-	init();
-	update_parameters();
-}
+	// Pre-condition: The path must not be an empty string
+	if (path.empty()) return;
```

Comment:
I changed this to check the string when the user returns from the file selection menu. There is now a warning here as this function should not get empty paths.